### PR TITLE
feat(splunk-hec): add plugin & fix dependencies

### LINKED ISSUE #125: Migrate to official splunk plugin GEM
In our testing, we found that the current Splunk plugin is not officially supported by Splunk and the maintainer is not actively working on it. 

I would like to propose to migrate to the official Splunk plugin 
https://github.com/splunk/fluent-plugin-splunk-hec

I am more than happy to submit the PR for the same.
@dimalbaby Thoughts ?

### PR DESCRIPTION
**Pull request by jliao2011**
Add the offical v1.2.4 splunk-hec plugin to rubygems
Removes fluent-plugin-logzio plugin due to dependency errors and deprecated plugin that hasn't been updated/maintained since March 2019
Updates fluent-plugin-loggly to v.1.0.pre to fix dependency conflict
Updates fluent-plugin-datadog to v0.12.1 to fix dependency conflict
Fixes #125

Signed-off-by: Jeff Liao <jliao.2015@gmail.com>

### CODE REVIEW

File: base-image/Gemfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ gem 'fluent-plugin-amqp', "0.14.0"
 gem 'fluent-plugin-azure-loganalytics', "0.7.0"
 gem 'fluent-plugin-cloudwatch-logs', "0.8.0"
```
Comment (kkapoor1987) [inline base-image/Gemfile:9]: @jliao2011 Can we also update fluent-plugin-cloudwatch-logs to latest version `v0.13.4` ?

**Comment by carpenterm** [comment]
@jliao2011 Thanks for your contribution, there are teams using the Logz.io plugin in production so we cannot just remove this entirely. Looks like Logz.io are working on an update - https://github.com/logzio/fluent-plugin-logzio/issues/33

**Comment by carpenterm** [comment]
@jliao2011 a new version of fluent-plugin-logzio (0.0.21) is available, please could you update your PR accordingly?

**Comment by jliao2011** [comment]
> @jliao2011 a new version of fluent-plugin-logzio (0.0.21) is available, please could you update your PR accordingly?

Done.

**Review by Cryptophobia — APPROVED** [review]
lgtm

**Pull request merged**